### PR TITLE
Integrate stable-24-3-9-cs-2 into stable-24-3

### LINKED ISSUE #9670: Integrate stable-24-3-9-cs-2 into stable-24-3
stable-24-3-9-cs-2 git hash 602dfb0787d4b33a0577f9572a94858b6aca157f

### PR DESCRIPTION
**Pull request by fexolm**
### Changelog entry 

Integrate all columnshard changes from stable-24-3-9-cs2 to stable-24-3
Basically, just rebasing stable-24-3.9-cs2(https://github.com/ydb-platform/ydb/commit/602dfb0787d4b33a0577f9572a94858b6aca157f) to stable-24.3

fixes #9670

### Changelog category 

* New feature
* Experimental feature
* Improvement
* Performance improvement
* Bugfix 
* Backward incompatible change
* Documentation (changelog entry is not required)
* Not for changelog (changelog entry is not required)

### Additional information

...

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -1 +1 @@
-* @ydb-platform/ReleaseApprovers
```
Comment (zverevgeny) [inline .github/CODEOWNERS:1]: please, revert this change

File: ydb/core/protos/table_service_config.proto [hunk 2]
```diff
@@ -285,7 +285,7 @@ message TTableServiceConfig {
     optional bool EnableOlapSink = 55 [default = false];
 
     optional bool EnablePerStatementQueryExecution = 56 [default = false];
-    optional bool EnableCreateTableAs = 57 [default = true];
```
Comment (zverevgeny) [inline ydb/core/protos/table_service_config.proto:288]: @nikvas0, check this change, please

**Review by nikvas0 — APPROVED** [review]
Kqp looks ok

**Review by fomichev3000 — APPROVED** [review]
